packages/storage/amplify_storage_s3/example/lib/main.dart

```diff
@@ -105,19 +108,249 @@ class HomeScreen extends StatefulWidget {
+    }
+  }
+
+  // download file on mobile
+  Future<void> downloadFileMobile(String key) async {
+    final documentsDir = await getApplicationDocumentsDirectory();
+    final filepath = '${documentsDir.path}/$key';
+    try {
+      await Amplify.Storage.downloadFile(
+        key: key,
+        localFile: AWSFile.fromPath(filepath),
+        onProgress: (p0) => _logger
+            .debug('Progress: ${p0.transferredBytes} / ${p0.totalBytes}%)'),
+      ).result;
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.error('Download error - ${e.message}');
+    }
+  }
+
+  // download file on web
+  Future<void> downloadFileWeb(String key) async {
+    try {
+      await Amplify.Storage.downloadFile(
+        key: key,
+        localFile: AWSFile.fromPath(key),
+      ).result;
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.error('Download error - ${e.message}');
+    }
+  }
+
+  // delete file from S3 bucket
+  Future<void> removeFile({
+    required String key,
+    required StorageAccessLevel accessLevel,
+  }) async {
+    try {
+      await Amplify.Storage.remove(
+        key: key,
+        options: StorageRemoveOptions(accessLevel: accessLevel),
+      ).result;
+      setState(() {
+        // set the imageUrl to empty if the deleted file is the one being displayed
+        imageUrl = '';
+      });
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.error('Delete error - ${e.message}');
+    }
+  }
+
+  // get the url of a file in the S3 bucket
+  Future<String> getUrl({
+    required String key,
+    required StorageAccessLevel accessLevel,
+  }) async {
+    try {
+      final result = await Amplify.Storage.getUrl(
+        key: key,
+        options: StorageGetUrlOptions(
+          accessLevel: accessLevel,
+          pluginOptions: const S3GetUrlPluginOptions(
+            validateObjectExistence: true,
+            expiresIn: Duration(minutes: 1),
+          ),
+        ),
+      ).result;
+      setState(() {
+        imageUrl = result.url.toString();
+      });
+      return result.url.toString();
+    } on StorageException catch (e) {
+      _logger.error('Get URL error - ${e.message}');
+      rethrow;
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         title: const Text('Amplify Storage Example'),
       ),
-      body: const Center(
-        child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            Text('Examples are coming...'),
-          ],
-        ),
+      body: Stack(
+        children: [
+          Center(
+            child: Padding(
+              padding: const EdgeInsets.all(20),
+              child: ListView(
+                children: [
+                  for (final item in list)
+                    ListTile(
+                      onTap: () => {
```

Comment:
Dart syntax is `() => single stmt fn` or `() { multi-stmt fn }`

The JS syntax of `() => { multi-stmt fn }` will compile sometimes in Dart but it means "a single-statement function which constructs a Set" which won't work as expected always.

So, both of the following are correct, but not what's currently written.

```dart
                      onTap: () {
                        getUrl(
                          key: item.key,
                          accessLevel: StorageAccessLevel.guest,
                        );
                      },
```

```dart
                      onTap: () => getUrl(
                        key: item.key,
                        accessLevel: StorageAccessLevel.guest,
                      ),
```